src/txrebroadcast.h

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) 2021 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef BITCOIN_TXREBROADCAST_H
+#define BITCOIN_TXREBROADCAST_H
+
+#include <policy/feerate.h>
+#include <tuple>
+#include <txmempool.h>
+#include <util/hasher.h>
+#include <util/time.h>
+#include <validation.h>
+
+#include <boost/multi_index/hashed_index.hpp>
+#include <boost/multi_index/member.hpp>
+#include <boost/multi_index/ordered_index.hpp>
+#include <boost/multi_index_container.hpp>
+
+struct TxIds {
```

Comment:
It seems unnecessary to define a type here. `GetRebroadcastTransactions()` could just return a vector of std::pairs, and the one caller (`UpdatedBlockTip()`) can unpack that vector of pairs using :sparkle: s t r u c t u t e d :sparkle: b i n d i n g s :sparkle:. Here's what the caller would look like:

```diff
diff --git a/src/net_processing.cpp b/src/net_processing.cpp
index 7811e3ca40..e5a425b4c1 100644
--- a/src/net_processing.cpp
+++ b/src/net_processing.cpp
@@ -1393,12 +1393,12 @@ void PeerManagerImpl::UpdatedBlockTip(const CBlockIndex *pindexNew, const CBlock
 
     // Rebroadcast selected mempool transactions
     if (gArgs.GetArg("-rebroadcast", DEFAULT_REBROADCAST_ENABLED)) {
-        std::vector<TxIds> rebroadcast_txs = m_txrebroadcast.GetRebroadcastTransactions();
+        auto rebroadcast_txs = m_txrebroadcast.GetRebroadcastTransactions();
 
         LOCK(cs_main);
 
-        for (auto ids : rebroadcast_txs) {
-            RelayTransaction(ids.m_txid, ids.m_wtxid, m_connman);
+        for (auto [txid, wtxid] : rebroadcast_txs) {
+            RelayTransaction(txid, wtxid, m_connman);
         }
     }
```

That client code works whether you return a vector of `TxIds` or a vector of `std::pair<uint256, uint256>>`s (which is as it should be :slightly_smiling_face: )

Reply:
yeah, I considered this approach but I went for an explicitly defined structure because its easy to mix up txid & wtxid, and doing so would cause quiet failures. It feels harder for a call site to accidentally mixup with the named access.